sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlockBlobApiTests.java

```diff
@@ -1565,6 +1566,41 @@ public void encryptionUploadISNoLengthWithOptions() {
         assertArraysEqual(randomData, os.toByteArray());
     }
 
+    @ParameterizedTest
+    @MethodSource("modifyUserAgentSupplier")
+    public void modifyUserAgent(EncryptionVersion encryptionVersion) {
+        EncryptedBlobClient ebc = new EncryptedBlobClient(mockAesKey(getEncryptedClientBuilder(fakeKey, null,
+            ENV.getPrimaryAccount().getCredential(), cc.getBlobContainerUrl(), encryptionVersion)
+            .blobName(generateBlobName())
+            .addPolicy(getUserAgentHeaderPolicy(encryptionVersion))
+            .buildEncryptedBlobAsyncClient()));
+
+        // the getUserAgentHeaderPolicy will check that the user agent is set correctly
+        ebc.uploadWithResponse(new BlobParallelUploadOptions(DATA.getDefaultInputStream()), null, null);
+    }
+
+    private static Stream<Arguments> modifyUserAgentSupplier() {
+        return Stream.of(Arguments.of(EncryptionVersion.V1), Arguments.of(EncryptionVersion.V2));
+    }
+
+    private static HttpPipelinePolicy getUserAgentHeaderPolicy(EncryptionVersion version) {
+        return new HttpPipelinePolicy() {
+            @Override
+            public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+                String userAgent = context.getHttpRequest().getHeaders().getValue(HttpHeaderName.USER_AGENT);
+                String expectedUserAgent = "azstorage-clientsideencryption/"
+                    + (version == EncryptionVersion.V2 ? "2.0" : "1.0");
+                assertTrue(userAgent.startsWith(expectedUserAgent));
```

Comment:
We should also add a test where the applicationId is set to ensure that the encryption version is correctly inserted between applicationId and the client library name.